nestkernel/connection_manager.cpp

```diff
@@ -1276,6 +1296,26 @@ nest::ConnectionManager::get_connections(
       source->toVector( sources );
       std::sort( sources.begin(), sources.end() );
 
+      std::vector< index > target_nodes;
+      std::vector< index > target_devices;
+      if ( target != 0 )
+      {
+        for ( size_t t_id = 0; t_id < target->size(); ++t_id )
+        {
+          const index target_gid = target->get( t_id );
+          if ( kernel()
+                 .node_manager.get_node( target_gid, tid )
+                 ->has_proxies() )
+          {
+            target_nodes.push_back( target_gid );
+          }
+          else
+          {
+            target_devices.push_back( target_gid );
+          }
+        }
+      }
```

Comment:
The code for splitting here is the same as above, isn't it? Can you refactor this into a method to avoid duplication?